include/SFML/Graphics/Rect.inl

```diff
@@ -22,125 +22,179 @@
 //
 ////////////////////////////////////////////////////////////
 
+////////////////////////////////////////////////////////////
+// Headers
+////////////////////////////////////////////////////////////
+#include <SFML/Graphics/Rect.hpp>
```

Comment:
Is the `Rect.hpp` include required here? We previously weren't including this in the `.inl` file, not sure why we should change that

Reply:
Because of the nature of the .inl files, Visual Studio didn't recognize basically anything. I made the syntax very much like a normal source file, so it's comfortable to read, and the IDE knows what's going on. The architecture didn't change at all, of course

Reply:
I don't think that just because Visual Studio fails to parse a `.inl` file that adding this include becomes worth it

Reply:
I only made this **smallest** refactor in the file that is in **my responsibility** area. It's literally **12 lines**, but it's **so much better**! 

We can't implement those methods in a source file, because they must be known at compile time, so we have to use this awkward semantic, but let's at least make it **look nice**!